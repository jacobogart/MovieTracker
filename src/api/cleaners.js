import { imageUrl } from '../utils/API-logins';

export const cleanAllMovies = (moviesArray) => {
  const cleanMovies = moviesArray.map(movie => {
    const { 
      id, 
      title, 
      poster_path, 
      overview, 
      release_date, 
      vote_average 
    } = movie;

    return { 
      id, 
      title, 
      posterImg: `${imageUrl}${poster_path}`, 
      overview, 
      releaseDate: release_date, 
      rating: vote_average 
    };
  });

  return cleanMovies;
};

export const cleanPeople = (data) => {
  const director = data.crew.filter(person => person.job === "Director");
  const writer = data.crew.filter(person => person.job === "Writer");
  const cast = data.cast.map(person => {
    let { name, character, profile_path,  cast_id} = person;

    return { 
      name, 
      character, 
      headshot: `${imageUrl}${profile_path}`, 
      id: cast_id 
    };
  });

  return { director, writer, cast };
};