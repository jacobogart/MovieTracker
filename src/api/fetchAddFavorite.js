import { serverUrl } from "./pathNames";

export const fetchAddFavorite = (cardData) => {
  const url = `${serverUrl}/favorites/new`;
  const { 
    user, 
    title, 
    overview, 
    posterImg, 
    id, 
    rating, 
    releaseDate 
  } = cardData;
  let poster_path = posterImg.split('500')[1];
  const body = {
    movie_id: id, 
    user_id: user.id,
    title, 
    poster_path,
    release_date: releaseDate, 
    vote_average: rating, 
    overview
  };
  const init = {
    method: 'POST',
    body: JSON.stringify(body),
    headers: { 'Content-Type': 'application/json' }
  };

  return fetch(url, init)
    .then(response => {
      if (!response.ok) {
        throw Error('Failed to add favorite');
      } else {
        return response.json();
      }
    });
};
