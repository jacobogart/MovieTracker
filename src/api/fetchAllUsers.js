import { serverUrl } from "./pathNames";

export const fetchAllUsers = () => {
  return fetch(serverUrl)
    .then(response => {
      if (!response.ok) {
        throw new Error("Failed to create account");
      } else {
        return response.json();
      }
    })
    .then(data => data.data);
};
