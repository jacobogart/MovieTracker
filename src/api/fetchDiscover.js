import { jacobKey, baseUrl } from '../utils/API-logins'

const fetchDiscover = () => {
  return fetch(`${baseUrl}discover/movie${jacobKey}`)
    .then(response => {
      if (!response.ok) {
        throw new Error('Discover failed to fetch');
      } else {
        return response.json();
      }
  });
}

export default fetchDiscover;

