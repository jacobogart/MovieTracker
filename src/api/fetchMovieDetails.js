import { jacobKey } from '../utils/API-logins';
import { baseUrl } from './pathNames';

const fetchMoviePeople = (id) => {
  return fetch(`${baseUrl}movie/${id}/credits${jacobKey}`)
    .then(response => {
      if (!response.ok) {
        throw new Error('People failed to fetch');
      } else {
        return response.json();
      }
    });
};

export default fetchMoviePeople;
