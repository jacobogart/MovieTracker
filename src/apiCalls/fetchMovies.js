import { jacobKey, baseUrl } from '../utils/API-logins'

export const fetchMovies = () => {
  fetch(`${baseUrl}discover/movie${jacobKey}`)
    .then(response => response.json())
    .then(result => console.log(result));
}

