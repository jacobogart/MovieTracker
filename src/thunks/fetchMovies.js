import { jacobKey } from '../utils/API-logins';
import { baseUrl } from '../api/pathNames.js';
import { addDiscoverMovies } from '../actions';
import { cleanAllMovies } from '../api/cleaners';

export const fetchMovies = (fetchCase, query) => {
  let url;
  
  switch(fetchCase) {
    case 'search':
      url = `${baseUrl}search/movie${jacobKey}&language=en-US&query=${query}&page=1&include_adult=false`
      break;
    case 'discover':
      url = `${baseUrl}discover/movie${jacobKey}`
      break;
    default:
      console.log('Need fetchCase');
  } 

  return async (dispatch) => {
    try {
      const response = await fetch(url);
      if (!response.ok) {
        throw new Error("Search failed");
      }
      const data = await response.json();
      const movies = cleanAllMovies(data.results);

      dispatch(addDiscoverMovies(movies));
    } catch(error) {
      console.log(error);
    }
  }
}