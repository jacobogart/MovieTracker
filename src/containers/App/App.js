import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Switch, Route } from 'react-router';
import  fetchDiscover  from '../../api/fetchDiscover';
import { cleanAllMovies } from '../../api/cleaners';
import { addDiscoverMovies } from '../../actions';
import Header from '../../components/Header/Header';
import MovieContainer from '../MovieContainer/MovieContainer';
import SignUpForm from '../SignUpForm/SignUpForm';

export class App extends Component {
  constructor() {
    super();
    this.state = {
      error: ''
    }
  }
  
  componentDidMount() {
    this.setDiscoverMovies();
  }

  setDiscoverMovies = () => {
    fetchDiscover()
      .then(response => cleanAllMovies(response.results))
      .then(movies => this.props.addDiscoverMovies(movies))
      .catch(error => this.setState({ error: error.message }));
  }
  
  render() {
    return(
      <div className="App">
        <Header />
        <Switch>
          <Route exact path="/" component={MovieContainer} />
          <Route exact path="/signup" component={SignUpForm} />
        </Switch> 
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  movies: state.movies
});

export const mapDispatchToProps = (dispatch) => ({
  addDiscoverMovies: (movies) => dispatch(addDiscoverMovies(movies))
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
