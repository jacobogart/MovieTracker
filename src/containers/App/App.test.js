import React from 'react';
import { shallow } from 'enzyme';
import { App, mapStateToProps, mapDispatchToProps } from './App';
import { cleanAllMovies } from '../../api/cleaners';
import { mockUncleanMovie } from '../../api/__tests__/mockData';
import { fetchMovies } from '../../thunks/fetchMovies';

jest.mock('../../thunks/fetchMovies');
jest.mock('../../api/cleaners.js');

describe('App', () => {
  let wrapper, instance;
  let mockSetMovies = jest.fn();
  
  beforeEach(() => {
    wrapper = shallow(<App setMovies={mockSetMovies}/>);
    instance = wrapper.instance();
  });

  afterEach(() => {
    mockSetMovies.mockClear();
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a defult state', () => {
    wrapper = shallow(<App />, { disableLifecycleMethods: true });
    expect(wrapper.state()).toEqual({ error: ''});
  });

  it('CDM should call setMovies method', () => {
    expect(mockSetMovies).toHaveBeenCalled();
  });
});
