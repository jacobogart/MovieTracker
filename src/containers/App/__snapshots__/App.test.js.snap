// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div
  className="App"
>
  <Header />
  <Switch>
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(MovieContainer)",
          "type": [Function],
        }
      }
      exact={true}
      path="/"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(MovieContainer)",
          "type": [Function],
        }
      }
      exact={true}
      path="/search"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(SignUpForm)",
          "type": [Function],
        }
      }
      exact={true}
      path="/signup"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(MovieDetailsContainer)",
          "type": [Function],
        }
      }
      path="/movie/:id"
    />
  </Switch>
  <Footer />
</div>
`;
